pkg/apis/pipeline/v1/pipeline_types.go

```diff
@@ -352,9 +353,52 @@ func (pt *PipelineTask) GetMatrixCombinationsCount() int {
 	if !pt.IsMatrixed() {
 		return 0
 	}
+	// Explicit combinations in the Matrix
+	// Calculate based on number of include Names and return count
+	if pt.Matrix.MatrixHasInclude() && !pt.Matrix.MatrixHasParams() {
+		var count = 0
+		// Iterate over each include.Name that has params
+		for _, include := range pt.Matrix.Include {
+			for _, includeParams := range include.Params {
+				// Increment count for every include name
+				if includeParams.Name != "" {
+					count++
+					break
+				}
+			}
+		}
+		return count
```

Comment:
why don't we just take the length of pt.Matrix.Include?

if we require that the name of a parameter is a non-empty string, then we can add a separate validation for that, instead of accounting for it in the count for matrix combinations

Reply:
Ok sounds good. Do you know if a "" param name is valid?

Reply:
I think that it should not be a valid param name, but I'm not sure if we have a validation for that -- could you check and, if it doesn't exist, create another PR proposing that validation?

Reply:
@jerop It doesn't look like we have any validation checking that a param name is not empty. I can open a separate PR to add this validation.
cc: @pritidesai @lbernick